codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/ClientSimpleMethods.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.codegen.poet.client;
+
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.FieldSpec;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.TypeSpec;
+import java.util.List;
+import javax.lang.model.element.Modifier;
+import software.amazon.awssdk.codegen.model.intermediate.IntermediateModel;
+import software.amazon.awssdk.codegen.model.intermediate.OperationModel;
+import software.amazon.awssdk.codegen.poet.ClassSpec;
+import software.amazon.awssdk.codegen.poet.PoetExtensions;
+import software.amazon.awssdk.codegen.poet.PoetUtils;
+import software.amazon.awssdk.regions.Region;
+
+public class ClientSimpleMethods implements ClassSpec {
+
+    private static final String APPROVED_VERBS = "(get|list|describe).*";
+
+    private final IntermediateModel model;
+    private final PoetExtensions poetExtensions;
+
+    public ClientSimpleMethods(IntermediateModel model) {
+        this.model = model;
+        this.poetExtensions = new PoetExtensions(model);
+    }
+
+    @Override
+    public TypeSpec poetSpec() {
+
+        ClassName className = poetExtensions.getClientClass("SimpleMethodsIntegrationTest");
+        ClassName interfaceClass = poetExtensions.getClientClass(model.getMetadata().getSyncInterface());
+        List<String> whitelistedApis = model.getCustomizationConfig().getVerifiedSimpleMethods();
+
+        TypeSpec.Builder builder = PoetUtils.createClassBuilder(className)
+                                            .addModifiers(Modifier.PUBLIC)
+                                            .addField(FieldSpec.builder(interfaceClass, "client")
+                                                               .addModifiers(Modifier.PRIVATE, Modifier.STATIC)
+                                                               .build())
+                                            .addMethod(setup());
+
+        model.getOperations().entrySet().stream()
+             .filter(v -> v.getValue().getInputShape().isSimpleMethod())
+             .filter(v -> !whitelistedApis.contains(v.getValue().getMethodName()))
```

Comment:
This filtering does two things:

First, only generate tests for simple methods.

Second, don't generate tests for whitelisted/verified simple methods. These are simple methods that have been verified to work but should not be tested likely due to creating resources.